components/brave_new_tab_ui/reducers/stack_widget_reducer.ts

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Reducer } from 'redux'
+import { types } from '../constants/stack_widget_types'
+
+const removeStackWidget = (widget: NewTab.StackWidget, state: NewTab.State): NewTab.State => {
+  let { removedStackWidgets, widgetStackOrder } = state
+
+  if (!widgetStackOrder.length) {
+    return state
+  }
+
+  if (!removedStackWidgets.includes(widget)) {
+    removedStackWidgets.push(widget)
+  }
+
+  state = {
+    ...state,
+    removedStackWidgets
+  }
+
+  return state
+}
+
+const setForegroundStackWidget = (widget: NewTab.StackWidget, state: NewTab.State): NewTab.State => {
+  let newWidgetStackOrder = state.widgetStackOrder
+
+  newWidgetStackOrder = newWidgetStackOrder.filter((stackWidget: NewTab.StackWidget) => {
```

Comment:
nit: we really don't want to mutate the state. You're not doing that here, but might be clearer to be explicit about old array vs new array. Something like
```
const oldWidgetStackOrder = state.widgetStackOrder
const newWidgetStackOrder = oldWidgetStackOrder.filter(...
```
...or similar...